robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowPendingIntent.java

```diff
@@ -134,37 +186,30 @@ public int getFlags() {
     return flags;
   }
 
+  public boolean isCanceled() {
+    return canceled;
+  }
+
+  @Implementation
+  public String getTargetPackage() {
+    return getCreatorPackage();
+  }
+
+  @Implementation(minSdk = 17)
+  public String getCreatorPackage() {
+    return (creatorPackage == null)
+        ? savedContext.getPackageName()
+        : creatorPackage;
+  }
+
+  public void setCreatorPackage(String creatorPackage) {
+    this.creatorPackage = creatorPackage;
+  }
+
   @Override
   @Implementation
   public boolean equals(Object o) {
-    if (this == o) return true;
```

Comment:
Ah interesting, I didn't see that more caching was added to the `PendingIntent#create` methods.